tests/book.test.js

```diff
@@ -0,0 +1,301 @@
+          genreId: genre.body.id,
+        });
+        expect(response.status).to.equal(400);
+        expect(response.body).to.equal("The book title cannot be empty");
+      });
+
+      it("authorId must be exist", async () => {
+        const response = await request(app).post("/books").send({
+          title: "title001",
+          ISBN: "isbn001",
+          genreId: genre.body.id,
+        });
+        expect(response.status).to.equal(400);
+        expect(response.body).to.equal("This Book must have an author");
+      });
+
+      it("authorId must not be empty", async () => {
+        const response = await request(app).post("/books").send({
+          title: "title001",
+          ISBN: "isbn001",
+          authorId: "",
+          genreId: genre.body.id,
+        });
+        expect(response.status).to.equal(400);
+        expect(response.body).to.equal("This Book must have an author");
+      });
+
+      it("authorId must be existed in Authors table", async () => {
+        const response = await request(app).post("/books").send({
+          title: "title001",
+          ISBN: "isbn001",
+          authorId: 999,
+          genreId: genre.body.id,
+        });
+        expect(response.status).to.equal(400);
+        expect(response.body).to.equal(
+          'Key (authorId)=(999) is not present in table "Authors".'
+        );
+      });
+    });
+  });
+
+  describe("with records in the database", () => {
+    let books;
+
+    beforeEach(async () => {
+      books = await Promise.all([
+        Book.create({
+          title: "title001",
+          authorId: 1,
+          genreId: 1,
+          ISBN: "isbn001",
+        }),
+        Book.create({
+          title: "title002",
+          authorId: 2,
+          genreId: 2,
+          ISBN: "isbn002",
+        }),
+        Book.create({
+          title: "title006",
+          authorId: 2,
+          genreId: 3,
+          ISBN: "isbn006",
+        }),
+        Book.create({
+          title: "title003",
+          authorId: 3,
+          genreId: 3,
+          ISBN: "isbn003",
+        }),
+        Book.create({
+          title: "title004",
+          authorId: 4,
+          genreId: 3,
+          ISBN: "isbn004",
+        }),
+        Book.create({
+          title: "title005",
+          authorId: 5,
+          genreId: 3,
+          ISBN: "isbn005",
+        }),
+      ]);
+    });
+
+    describe("GET /books", () => {
+      it("gets all books records", async () => {
+        const response = await request(app).get("/books");
+        expect(response.status).to.equal(200);
+        expect(response.body.length).to.equal(6);
+
+        response.body.forEach((book) => {
+          const expected = books.find((a) => a.id === book.id);
+          expect(book.title).to.equal(expected.title);
+          expect(book.authorId).to.equal(expected.authorId);
+          expect(book.genreId).to.equal(expected.genreId);
+          expect(book.ISBN).to.equal(expected.ISBN);
+        });
```

Comment:
It would be great if here (and your get by id requests) it would test to see if eager loading is working correcting. i.e. is the actual genre and author name being returned too, as well as the ids